cognite/client/_api/hosted_extractors/destinations.py

```diff
@@ -0,0 +1,272 @@
+from __future__ import annotations
+
+from collections.abc import Iterator
+from typing import TYPE_CHECKING, Any, Sequence, overload
+
+from cognite.client._api_client import APIClient
+from cognite.client._constants import DEFAULT_LIMIT_READ
+from cognite.client.data_classes.hosted_extractors.destinations import (
+    Destination,
+    DestinationList,
+    DestinationUpdate,
+    DestinationWrite,
+)
+from cognite.client.utils._experimental import FeaturePreviewWarning
+from cognite.client.utils._identifier import IdentifierSequence
+from cognite.client.utils.useful_types import SequenceNotStr
+
+if TYPE_CHECKING:
+    from cognite.client import ClientConfig, CogniteClient
+
+
+class DestinationsAPI(APIClient):
+    _RESOURCE_PATH = "/hostedextractors/destinations"
+
+    def __init__(self, config: ClientConfig, api_version: str | None, cognite_client: CogniteClient) -> None:
+        super().__init__(config, api_version, cognite_client)
+        self._warning = FeaturePreviewWarning(
+            api_maturity="alpha", sdk_maturity="alpha", feature_name="Hosted Extractors"
+        )
+        self._CREATE_LIMIT = 100
+        self._LIST_LIMIT = 100
+        self._RETRIEVE_LIMIT = 100
+        self._DELETE_LIMIT = 100
+        self._UPDATE_LIMIT = 100
+
+    @overload
+    def __call__(
+        self,
+        chunk_size: None = None,
+        limit: int | None = None,
+    ) -> Iterator[Destination]: ...
+
+    @overload
+    def __call__(
+        self,
+        chunk_size: int,
+        limit: int | None = None,
+    ) -> Iterator[Destination]: ...
+
+    def __call__(
+        self,
+        chunk_size: int | None = None,
+        limit: int | None = None,
+    ) -> Iterator[Destination] | Iterator[DestinationList]:
+        """Iterate over destinations
+
+        Fetches Destination as they are iterated over, so you keep a limited number of spaces in memory.
+
+        Args:
+            chunk_size (int | None): Number of Destinations to return in each chunk. Defaults to yielding one Destinatio a time.
+            limit (int | None): Maximum number of Destination to return. Defaults to returning all items.
```

Comment:
**Note 1**: 10 other occurrences of `space(s)`
**Note 2**: We don't capitalize the resource except in the returns section
```suggestion
        Fetches destinations as they are iterated over, so you keep a limited number of destinations in memory.

        Args:
            chunk_size (int | None): Number of destination to return in each chunk. Defaults to yielding one Destination a time.
            limit (int | None): Maximum number of destination to return. Defaults to returning all items.
```